# `rfqvendor` — one row per bidder submission against an RFQ

Unity Catalog: `ingestion_framework_test.bid_data_exploration.rfqvendor`

Expected: the bidder-level record (maps to AGPOWER / AL Geemi / etc. in the sample data). Look for: bidder name/vendor id, round/revision number, submission date, total price if rolled up here, and status (submitted / disqualified / awarded).

## Run 1 — initial exploration ✅ *(completed — results confirmed, see FINDINGS.md)*

In [0]:
%sql
DESCRIBE TABLE ingestion_framework_test.bid_data_exploration.rfqvendor

col_name,data_type,comment
RFQNUM,varchar(20),null
VENDOR,varchar(44),null
CONTACT,varchar(240),null
PHONE,varchar(26),null
FAXPHONE,varchar(26),null
EMAIL,varchar(60),null
CUSTOMERNUM,varchar(25),null
FOB,varchar(25),null
FREIGHTTERMS,varchar(50),null
SHIPVIA,varchar(25),null


In [0]:
%sql
SELECT COUNT(*) AS row_count FROM ingestion_framework_test.bid_data_exploration.rfqvendor

row_count
412896


In [0]:
%sql
SELECT * FROM ingestion_framework_test.bid_data_exploration.rfqvendor LIMIT 20

RFQNUM,VENDOR,CONTACT,PHONE,FAXPHONE,EMAIL,CUSTOMERNUM,FOB,FREIGHTTERMS,SHIPVIA,PAYMENTTERMS,CURRENCYCODE,EXCHANGERATE,EXCHANGEDATE,BUYAHEAD,INTERNAL,REPLIEDDATE,GLCREDITACCT,INCLUSIVE1,INCLUSIVE2,INCLUSIVE3,INCLUSIVE4,INCLUSIVE5,VENDORQUOTENUM,ORGID,SITEID,RFQVENDORID,LANGCODE,HASLD,ROWSTAMP,BUYERREMARK,LINENUM,RFQV1,PRODUCTGROUPID,CURRENCYCODE2,RFQV3,RFQV4,DISCOUNT,SQR,MNGREMARK,ENTERDATE,RFQV2,OTHERCHG,RFQV5,SSUPREMARK,ENTEREDBY,ISVENINC,BYRRMK,SNSRMK,MGTRMK,BIDSTATUS,BIDSTATUSDATE,PASSWORD,PRICEIMPACT,REGRETREASON,RESETPASSWORD,CI,TCI,TI,USERID,VALIDTO,SUPPORTREMARKS,SUPPOERTCONTACTED,CIINITIATED,VALIDITYPERIOD,SCORE,SURVEYCREATED,BASETOTALAWARDCOST,TOTALAWARDCOST,TOTALDISCOUNT,TOTAWDWODISCOUNT,TOTBIDCOSTWDIS,TOTBIDCOSTWODIS,TOTALAWARDCOSTWITHTAX,COMPEVALTYPE,APPLYDISCOUNTBEFOREBIDS,APPLYDISCOUNTAFTERBIDS,DISCOUNT_PERCENT,APPLYDISCOUNT,DISCOUNTAFTERBIDS,APPLYDISCOUNTAFTERPI,DCI,DCIREQUIRED,DISCOUNTSTATUS,DISCOUNT_APPLIED_AFTERBIDS,DISCOUNT_APPLIED_AFTERPI,DISCOUNT_APPLIED_BEFOREBIDS,POSTBID_DISCOUNT_COUNTER,POSTBID_DISCOUNT_SENT,DISCOUNT_REVISION,DISCOUNT_SUBMISSION_DATE,TOTALAWARDCOSTWDIS,TOTALAWARDCOSTWITHTAXWDIS,DISCOUNT_APPLY_DATE,QUOTERENEWED,QUOTEVALIDITY,RFQV_EXTRA1,RFQV_EXTRA2,ISAWARDED,AGREECOC,WAIVEOWNERVAL
D1752-R1,001052,YOUSEF ALNOWAIS,02-6447373,02-6444066,mideast@emirates.net.ae,null,null,null,null,null,AED,null,null,0E-10,0E-10,null,XD-0000-00000-251040-00-000000000-00-000000000,1.0000000000,0E-10,0E-10,0E-10,0E-10,null,ADDCORG,ADDC,30422.0000000000,EN,0E-10,6716862368,null,null,null,null,null,1.0000000000,null,null,null,null,null,null,null,null,null,null,0E-10,null,null,null,null,null,null,0E-10,null,0E-10,0E-10,0E-10,0E-10,null,null,null,null,0E-10,null,null,1.0000000000,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,null,null,0E-10,0E-10,null,null,0E-10,0E-10,0E-10,0E-10,null,0E-10,0E-10,0E-10,null,0E-10,null,null,0.00,0.00,null,0E-10,null,null,null,0E-10,0E-10,0E-10
D1752-R1,001377,K . VISWANATHAN,02-6414093,02-6779257,null,null,null,null,null,null,AED,null,null,0E-10,0E-10,2003-12-17T09:02:27Z,XD-0000-00000-251040-00-000000000-00-000000000,1.0000000000,0E-10,0E-10,0E-10,0E-10,null,ADDCORG,ADDC,30423.0000000000,EN,0E-10,6716862369,null,null,null,null,null,1.0000000000,null,null,null,null,null,null,null,null,null,null,0E-10,null,null,null,null,null,null,0E-10,null,0E-10,0E-10,0E-10,0E-10,null,null,null,null,0E-10,null,null,1.0000000000,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,null,null,0E-10,0E-10,null,null,0E-10,0E-10,0E-10,0E-10,null,0E-10,0E-10,0E-10,null,0E-10,null,null,0.00,0.00,null,0E-10,null,null,null,0E-10,0E-10,0E-10
D1752-R1,001562,KNR PILLAI,02-6723732,02-6782496,null,null,null,null,null,null,AED,null,null,0E-10,0E-10,2003-12-17T09:03:22Z,XD-0000-00000-251040-00-000000000-00-000000000,1.0000000000,0E-10,0E-10,0E-10,0E-10,null,ADDCORG,ADDC,30424.0000000000,EN,0E-10,6716862370,null,null,null,null,null,1.0000000000,null,null,null,null,null,null,null,null,null,null,0E-10,null,null,null,null,null,null,0E-10,null,0E-10,0E-10,0E-10,0E-10,null,null,null,null,0E-10,null,null,1.0000000000,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,null,null,0E-10,0E-10,null,null,0E-10,0E-10,0E-10,0E-10,null,0E-10,0E-10,0E-10,null,0E-10,null,null,0.00,0.00,null,0E-10,null,null,null,0E-10,0E-10,0E-10
D1752-R1,002337,MAIN,02-6339591,02-6317066,null,null,null,null,null,null,AED,null,null,0E-10,0E-10,null,XD-0000-00000-251040-00-000000000-00-000000000,1.0000000000,0E-10,0E-10,0E-10,0E-10,null,ADDCORG,ADDC,30425.0000000000,EN,0E-10,6716862371,null,null,null,null,null,1.0000000000,null,null,null,null,null,null,null,null,null,null,0E-10,null,null,null,null,null,null,0E-10,null,0E-10,0E-10,0E-10,0E-10,null,null,null,null,0E-10,null,null,1.0000000000,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,CGTqea+foUw=,null,null,0E-10,0E-10,null,null,0E-10,0E-10,0E-10,0E-10,null,0E-10,0E-10,0E-10,null,0E-10,null,null,0.

## Run 2 — follow-up queries ✅ *(completed — results documented in FINDINGS.md)*

First pass findings (full write-up in `databricks/FINDINGS.md`):
- **412,896 rows** vs. `rfq`'s 30,338 — ~13.6 invited-vendor rows per RFQ on average. This is invitees, not just actual submitters.
- **`VENDOR` is a numeric code (`001052`), not a company name.** There's no vendor/company master table in our current 7 — need to find one to resolve codes to names like "AGPOWER"/"AL Geemi".
- **`RFQVENDORID` is this table's PK.** Almost certainly the FK `quotationline` points back with — pending `quotationline`'s `DESCRIBE` to confirm the exact column name (case matters for the underscore: is it `RFQVENDORID` or `RFQVENDOR_ID`?).
- **Sample `RFQNUM` values have a `-R1` suffix** (`D1752-R1`) — this may mean negotiation rounds are separate RFQ *records* (a new RFQNUM per round), not a round column on one RFQ. That would conflict with the `DISCOUNT_REVISION` theory from `rfq` — there may be two different "round" concepts (formal re-tender vs. post-bid discount loop). Needs clarifying with TAQA.
- **7 award/discount-cost columns are typed `binary`, not decimal** (`TOTALAWARDCOST`, `TOTALDISCOUNT`, `TOTBIDCOSTWDIS`, etc.) and every sample row shows the *same* value (`CGTqea+foUw=`) for all of them — looks encrypted/masked, not just null. Two proper decimal columns exist alongside them (`TOTALAWARDCOSTWDIS`, `TOTALAWARDCOSTWITHTAXWDIS`) but are null in this (2003-era) sample — worth checking if those are populated on recent rows instead.

### Does `RFQNUM` really carry a round suffix?

In [0]:
%sql
SELECT regexp_extract(RFQNUM, '-(R[0-9]+)$', 1) AS round_suffix, COUNT(*) AS n
FROM ingestion_framework_test.bid_data_exploration.rfqvendor
GROUP BY round_suffix
ORDER BY n DESC

round_suffix,n
,412386
R1,499
R2,11


### Bid status distribution

In [0]:
%sql
SELECT BIDSTATUS, COUNT(*) AS n FROM ingestion_framework_test.bid_data_exploration.rfqvendor
GROUP BY BIDSTATUS ORDER BY n DESC

BIDSTATUS,n
null,221159
SUBMITTED,61766
NOT SIGNED,49862
REGRETTED,31467
COLLECTED,28750
NOT COLLECTED,14094
NOT QUOTED,3629
PI SUBMITTED,2023
WITHDRAWN,146


### Is `DISCOUNT_REVISION` / `POSTBID_DISCOUNT_COUNTER` populated at the vendor level?

In [0]:
%sql
SELECT DISCOUNT_REVISION, POSTBID_DISCOUNT_COUNTER, COUNT(*) AS n
FROM ingestion_framework_test.bid_data_exploration.rfqvendor
GROUP BY DISCOUNT_REVISION, POSTBID_DISCOUNT_COUNTER
ORDER BY n DESC
LIMIT 30

DISCOUNT_REVISION,POSTBID_DISCOUNT_COUNTER,n
null,null,400386
1.0000000000,1.0000000000,3834
1.0000000000,null,3263
2.0000000000,2.0000000000,1892
3.0000000000,3.0000000000,878
2.0000000000,1.0000000000,778
2.0000000000,null,490
3.0000000000,2.0000000000,284
4.0000000000,4.0000000000,274
3.0000000000,1.0000000000,138


### Are the proper-decimal award-cost columns populated on recent rows?

In [0]:
%sql
SELECT RFQNUM, VENDOR, ENTERDATE, TOTALAWARDCOSTWDIS, TOTALAWARDCOSTWITHTAXWDIS, SCORE, BIDSTATUS
FROM ingestion_framework_test.bid_data_exploration.rfqvendor
WHERE ENTERDATE >= '2024-01-01' AND TOTALAWARDCOSTWDIS IS NOT NULL
ORDER BY ENTERDATE DESC
LIMIT 30

RFQNUM,VENDOR,ENTERDATE,TOTALAWARDCOSTWDIS,TOTALAWARDCOSTWITHTAXWDIS,SCORE,BIDSTATUS
N-20821,99642847,2026-07-14T22:41:07Z,0.00,0.00,null,SUBMITTED
N-20821,99761819,2026-07-08T08:41:39Z,0.00,0.00,null,SUBMITTED
N-20821,99527411,2026-07-06T10:11:45Z,0.00,0.00,null,SUBMITTED
A19892455,99440864,2026-07-06T09:49:41Z,0.00,0.00,null,SUBMITTED
A19892455,99128799,2026-07-03T12:51:36Z,0.00,0.00,null,SUBMITTED
A19892455,99519140,2026-07-03T11:36:19Z,0.00,0.00,null,SUBMITTED
A19892455,9932007,2026-07-03T10:29:10Z,0.00,0.00,null,SUBMITTED
A19892455,9909392,2026-07-03T10:04:43Z,0.00,0.00,null,SUBMITTED
A19892455,9924650,2026-07-03T10:01:54Z,0.00,0.00,null,SUBMITTED
A19892455,001332,2026-07-03T09:47:33Z,0.00,0.00,null,SUBMITTED


### Recent rfqvendor rows for a sanity check (any resembling D-111808?)

In [0]:
%sql
SELECT RFQNUM, VENDOR, CONTACT, BIDSTATUS, SCORE, ENTERDATE
FROM ingestion_framework_test.bid_data_exploration.rfqvendor
WHERE RFQNUM LIKE 'D-111808%'
ORDER BY ENTERDATE DESC

RFQNUM,VENDOR,CONTACT,BIDSTATUS,SCORE,ENTERDATE
D-111808,99473989,AHMAD JAMAL ANIS EL BALAA,SUBMITTED,null,2023-04-26T09:43:19Z
D-111808,99107175,A G DORAIRAJ,SUBMITTED,null,2023-04-26T09:37:24Z
D-111808,9909941,Syed Khader Basha,SUBMITTED,null,2023-04-26T08:50:18Z
D-111808,002599,Abubacker Siddique,SUBMITTED,null,2023-04-25T10:58:18Z
D-111808,003235,Subhash Zinjad,COLLECTED,null,2023-04-25T10:48:36Z
D-111808,99495736,JORDI OLIVA,SUBMITTED,null,2023-04-25T10:42:15Z
D-111808,001669,Shakun Goyal,REGRETTED,null,2023-04-25T09:22:01Z
D-111808,99141295,MOHANNAD SANDOQA,COLLECTED,null,2023-04-24T14:57:55Z
D-111808,001990,N Pavunraju,NOT SIGNED,null,2023-04-24T14:30:59Z
D-111808,99102876,Manesh Balakrishna Pillai,SUBMITTED,null,2023-04-24T14:18:06Z


**Observations (updated after first real run):**
- `RFQVENDORID` is the per-invited-vendor primary key; `VENDOR` is a vendor **code**, not a name — a vendor/company master table is needed and isn't among our current 7 tables.
- `RFQNUM` values like `D1752-R1` suggest rounds might be modeled as distinct RFQ records rather than a column — needs reconciling with `rfq`'s `DISCOUNT_REVISION` theory.
- Several award/discount-total columns are typed `binary` and look encrypted/masked in the sample rather than genuinely populated — flagged as a potential blocker for reading real pricing/award totals directly from this table.